### PR TITLE
[REVIEW] Cholesky rank one update prim

### PR DESCRIPTION
**Pull request by tfeher**
This PR implements rank one update for Cholesky decomposition. 

More info https://en.wikipedia.org/wiki/Cholesky_decomposition#Rank-one_update

This is a building block for LARS.

In addition to the new prim, cuBLAS wrappers for swap and copy are added.

### CODE REVIEW

**Review by dantegd — APPROVED** [review]
looks great

File: cpp/include/raft/linalg/cholesky_r1_update.cuh [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    stream);
+
+  // Check for numeric error: the above sqrt can have negative argument if the
+  // matrix is not positive definite or very ill conditioned.
+  math_t L_22_host;
+  raft::update_host(&L_22_host, L_22, 1, stream);
+  CUDA_CHECK(cudaStreamSynchronize(stream));
+  if (std::isnan(L_22_host)) {
+    THROW("Error during Cholesky rank one update");
+  }
```
Comment (teju85) [inline cpp/include/raft/linalg/cholesky_r1_update.cuh:196]: nitpick: this could be done via `ASSERT` macro for simplicity, right?

**Review by teju85 — CHANGES_REQUESTED** [review]

File: cpp/include/raft/linalg/cholesky_r1_update.cuh (hunk 1 above)
Comment (tfeher) [reply cpp/include/raft/linalg/cholesky_r1_update.cuh]: Yes, thanks for pointing this out! I have replaced it with assert.

**Review by teju85 — APPROVED** [review]
Changes LGTM.

**Comment by dantegd** [comment]
rerun tests

**Comment by dantegd** [comment]
rerun tests

**Pull request merged**